src/scanner.c

```diff
@@ -0,0 +1,310 @@
+#include <assert.h>
+#include <string.h>
+#include <tree_sitter/parser.h>
+#include <wctype.h>
+
+enum TokenType { COMMENT };
+
+typedef struct {
+  size_t quoted_string_id_length;
+  size_t quoted_string_id_capacity;
+  char *quoted_string_id;
+} Scanner;
```

Comment:
I'm not a big fan of this struct being called `Scanner`, it sounds confusing to me because this is not a tree-sitter scanner. It would rather have it called something like `quoted_string_id`

Reply:
In [tree-sitter-ocaml](https://github.com/tree-sitter/tree-sitter-ocaml/blob/fa4a8175a005a35c934ab36c4e4887d217061b2e/common/scanner.h) it contains the complete state of the scanner. Due to some simplifications, it's just `quoted_string_id` here. It can be simplified even more, by removing `scan_left_quoted_string_delimiter()` and `scan_right_quoted_string_delimiter()` (they will never be called in isolation here), and making the struct just a variable-length string that will be initialized in `scan_quoted_string()`. But that will make it more difficult to sync future changes to the tree-sitter-ocaml implementation, so I opted for keeping the diff as small as possible.

Reply:
No, I think that having `scan_{left,right}_quoted_string_delimiter` aside makes things clearer, so let's leave it this way.
And I get why this is called `Scanner` now, I totally buy the argument, thanks for the explanation.